### ISSUE TITLE
what is "_DUX4_patched " at the end of the test count file?

### ISSUE THREAD

**Issue by Mercilena** [gu-lab20/ShinyALL#2]

**Comment by gu-lab20** [comment]
In our pipeline, we specifically rescued the counts for DUX4 gene because DUX4 naturally has multiple copies in the genome and would lower the mapping quality and underestimate the reads for the DUX4 gene.  We have a script to do this job after the running of HTseqcount. But the B-ALL subtyping pipeline should work fine without this extra job.

We have added a new folder: https://github.com/gu-lab20/MD-ALL/tree/master/DUX4_patch to include the scripts for this purpose.

**Comment by Mercilena** [comment]
I ran the script with this command:   "./HTSeqDUX4patch.pl -f Homo_sapiens.GRCh38.V102.withChr.Exon.DUX4.bed -b F1_STAR_HG38_sorted.bam -h F1.HTSeq -o F1.HTSeq.DUX4patch"

This error appears and it seems to need ".bai" file too:

Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 1.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 1.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 1.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 2.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 2.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 2.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 3.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 3.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 3.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 4.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 4.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 4.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 5.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 5.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 5.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 6.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 6.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 6.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
Use of uninitialized value $chr in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 7.
Use of uninitialized value $start in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 7.
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 7.
[E::hts_open_format] Failed to open file "F1_STAR_HG38_sorted.bam" : No such file or directory
samtools view: failed to open "F1_STAR_HG38_sorted.bam" for reading: No such file or directory
sh: 1: cannot open !DOCTYPE:html: No such file
sh: 1: cannot open html:lang=en-data-color-mode=auto: No such file
Use of uninitialized value $end in concatenation (.) or string at ./HTSeqDUX4patch.pl line 28, <IN> line 10.
sh: 1: cannot open head: No such file
sh: 1: cannot open meta:charset=utf-8: No such file
sh: 1: Syntax error: end of file unexpected
sh: 1: Syntax error: end of file unexpected
sh: 1: Syntax error: end of file unexpected
sh: 1: Syntax error: end of file unexpected
sh: 1: cannot open link:rel=preconnect-href=https://github.githubassets.com: No such file
sh: 1: Syntax error: end of file unexpected

**Comment by gu-lab20** [comment]
Yes. It will need the bai file since the underlying code is based on samtools to select the reads within DUX4 gene regions.
